src/main/scala/is/hail/rvd/OrderedRVDPartitioner.scala

```diff
@@ -8,66 +8,144 @@ import org.apache.spark.Partitioner
 class OrderedRVDPartitioner(
   val numPartitions: Int,
   val partitionKey: Array[String], val kType: TStruct,
-  // rangeBounds is partition max, sorted ascending
-  // rangeBounds: Array[pkType]
   val rangeBounds: UnsafeIndexedSeq) extends Partitioner {
-  require((numPartitions == 0 && rangeBounds.isEmpty) || numPartitions == rangeBounds.length + 1,
-    s"nPartitions = $numPartitions, ranges = ${ rangeBounds.length }")
+  require(numPartitions == rangeBounds.size, s"nPartitions = $numPartitions, ranges = ${ rangeBounds.size }")
```

Comment:
I think just remove `numPartitions` now.  We had both because before, if rangeBounds was empty, we didn't know if there were 0 or 1 partitions.